### PR TITLE
Add operator EC2-kind E2E tests

### PR DESCRIPTION
**Pull request by fanny-jiang**
### What does this PR do?

Adds basic E2E tests for the operator binary deployment to a kind cluster on EC2.

* Provision kind cluster on EC2 VM
* Install operator with pulumi kustomize
* Deploy minimal DDA and check agent resources

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### Minimum Agent Versions

Are there minimum versions of the Datadog Agent and/or Cluster Agent required?

* Agent: vX.Y.Z
* Cluster Agent: vX.Y.Z

### Describe your test plan
See updated docs: https://github.com/DataDog/datadog-operator/pull/1102/files#diff-d5619710a7a4e352877e5406fdfdfe8ea91dfa1100651402644ab4304403f90c

### Checklist

- [x] PR has at least one valid label: `bug`, `enhancement`, `refactoring`, `documentation`, `tooling`, and/or `dependencies`
- [x] PR has a milestone or the `qa/skip-qa` label

### CODE REVIEW

File: docs/how-to-contribute.md [hunk 1]
```diff
@@ -75,3 +75,60 @@ kubectl -n $KUBE_NAMESPACE  apply -f examples/datadogagent/v2alpha1/min.yaml
...
+### End-to-End Tests
+
+The Datadog Operator end-to-end (E2E) tests run on [Pulumi][pulumi]-deployed test infrastructures, defined as "stacks". The test infrastructures are deployed using the [`test-infra-definitions`][test-infra-repo] and [`datadog-agent`][agent-e2e-source] E2E frameworks.
+
+**Prerequisites**
+
+Internal Datadog users may run E2E locally after completing the following prerequisites:
+
+* Access to the AWS `agent-sandbox` account
+* AWS keypair with your public ssh key created in the `agent-sandbox` account
```
Comment (rtrieu) [inline docs/how-to-contribute.md:104]:
```suggestion
* AWS keypair with your public SSH key created in the `agent-sandbox` account
```

**Review by rtrieu — APPROVED** [review]
Minor suggestion but LGTM

File: .gitignore [hunk 2]
```diff
@@ -100,3 +100,4 @@ tags
 .idea/
 # test errors
 errors
+go.work*
```
Comment (clamoriniere) [inline .gitignore:103]: can we make the  golang workspace (`go.work`) permanent?

Comment (fanny-jiang) [reply .gitignore:103]: I tried committing `go.work`, but it causes `make build` to fail. I think we'll need to investigate it separately and might require updating dependencies.

**Review by levan-m — APPROVED** [review]
nit - `common_test.go` could be rename since there are no tests just utilities. Otherwise looks good and work in CI 🚀

**Pull request merged**